Feature/53 odds and ends

chrisdicaprio opened the pull request:
closes #53

chrisbc commented on nzshm_common/location/__init__.py:
@chrisdicaprio FYI I'm finding it very helpful to add docstrings into the package __init__.py 

I think it makes sense to provide examples for the most commonly used features of that package. As you're hoisting these function in the API,  these docstrings are more important/valuable.

 ..eg https://github.com/GNS-Science/solvis/pull/48/files#diff-6babf3be0dc2af4e9ac592e409a3fcaee784fdffcb84dd9935f5eb0456c0cb57

chrisbc commented on nzshm_common/location/location.py:
```diff
@@ -63,7 +63,7 @@
 
 
-def _lat_lon(_id) -> Optional[LatLon]:
+def lat_lon_by_id(_id) -> Optional[LatLon]:
```
@chrisdicaprio I think a docstring is needed here as we're adding to the API. Can you please check it's being picked up in mkdocs also?

chrisbc commented on nzshm_common/location/location.py:
```diff
@@ -63,7 +63,7 @@
 
 
-def _lat_lon(_id) -> Optional[LatLon]:
+def lat_lon_by_id(_id) -> Optional[LatLon]:
```
@chrisdicaprio actually I'm confused why we'd elevate this _method anyway? -  isn't the preferred way the location_by_id() function??

chrisdicaprio replied:
I think we're both wrong. I should be using `get_locations`. The function name isn't obvious for the application, but it will return a list of `CodedLocations` given a list of location ids. This is used in `runzi` `toshi-hazard-post`, and the TS1170 lib.

chrisbc reviewed (commented):
@chrisdicaprio this PR raised a few questions for me are we elevating more than we need to?? ... recall the work the Zoey / David did on the location_by_id function. 

Maybe putting the some example docstrings to show how they're meant to be used will clarify this.

chrisdicaprio commented:
I have removed an import from the top level `__init__.py`:
```
from .location import location
```

and added an import of useful functions to the location/__init__.py
```
from .location import get_location_list, get_location_list_names, get_locations, location_by_id
```

This allows the user to do:
```
>>> from nzshm_common.location import location_by_id
```
but no longer allows
```
>>> from nzshm_common import location
>>> location.LOCATION_LISTS
```

The intent is to elevate the user API to the `nzshm_common.location` package and hide attributes of `nzshm_common.location.location` that the users is meant to access via helper functions rather than directly.

chrisbc reviewed (approved):
Thanks @chrisdicaprio, looks great
